lib/exabgp/application/healthcheck.py

```diff
@@ -472,7 +465,8 @@ def exabgp(target):
             setup_ips(options.ips, options.label, options.sudo)
 
         logger.info("send announces for %s state to ExaBGP", target)
-        metric = vars(options).get("{0}_metric".format(str(target).lower()))
+        if target != states.EXIT:
+            metric = vars(options).get("{0}_metric".format(str(target).lower()))
```

Comment:
I don't think this is needed. Mostly, `states.EXIT` is like `states.DOWN` with `options.withdraw_on_down` enabled. So, below, you should do `if options.withdraw_on_down or states.EXIT`.

Reply:
when exit and withdraw route， no need to change metric， but when announce， metric is needed

Reply:
This is handled below. As long as you change `if options.withdraw_on_down` to `if options.withdraw_on_down or target == states.EXIT`, `command` will be `withdraw` and metric and other stuff will not be used.

Reply:
event options.withdraw_on_down is True, command still can be “announce” or “withdraw”， metric is still needed

Reply:
Below, if you do (just below):

```python
        for ip in options.ips:
            if options.withdraw_on_down or target == states.EXIT:
                command = "announce" if target is states.UP else "withdraw"
            else:
                command = "announce"
```

When you are in exit state, command will be `"withdraw"` and metric and other stuff is ignored. No need to add `if target == states.EXIT` elsewhere.

Reply:
howerver the followings code will always be  executed, and will cause error since there is no EXIT_metirc
`metric = vars(options).get("{0}_metric".format(str(target).lower()))`
`metric += options.increase`

Reply:
You are right. I would suggest to use `vars(options).get("{0}_metric".format(str(target).lower()), 0)` to avoid this without litering the code with additional `if`.

Reply:
code updated, but  routes still can not be withdrawed.